### PR TITLE
[PE-4174] Added Enterprise Global Navigation example to Sidenav component

### PR DESCRIPTION
**Pull request by janibolkvadze**
### Commit summary

The first implementation of Enterprise Global Navigation 3rd level tabs.

- [x] Implemented 3rd level navigation with animation
- [x] Added an example to /javascript/sidenav view
- [x] Fixed some JS linting errors

Thoughts about the approach, class namings, etc? @mattnickles @jllr @dani-cl-madrid 

https://ctl.atlassian.net/browse/PE-4174

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/538/ for more details.

**Comment by janibolkvadze** [comment]
- [x] Updated Enterprise Global Navigation Sidenav design by overriding default properties
- [x] Fixed sliding border bug on inversed sidenavs
- [x] Improved closing behavior when subtabs are clicked
- [x] Improved activating of menu items when subtabs are visible

Still pending:
- [ ] Changing sliding border color from green to white when sliding from active menu item to non-active (and vice versa)

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/540/ for more details.

File: src/chi/javascript/components/sidenav.js [hunk 1]
```diff
@@ -192,10 +198,27 @@ class Sidenav extends Component {
     if (activator && drawer.id) {
       menuItemLink = this._getMenuItemLink(drawer);
       menuItem = menuItemLink.parentNode;
-      this.close(menuItemLink);
+      if (drawerMenuItem.querySelector(`.${DRAWER_ITEM_LIST_CLASS}`) === null ||
+        e.target.classList.contains('a-sidenav__drawer-item-tab')) {
+        this.close(menuItemLink);
+      }
 
       if (drawerMenuItem) {
```
Comment (mattnickles) [inline src/chi/javascript/components/sidenav.js]: Can the Drawer menu item list function as a toggle? If expanded, clicking again would collapse

File: src/chi/components/sidebar/sidenav.scss [hunk 2]
```diff
@@ -376,6 +404,114 @@ $sidenav-lg-width: 12.5rem;
       }
     }
   }
+
+  &.-global-nav {
+    &.-inverse {
+      .m-sidenav__content {
+        border-right: 0;
```
Comment (mattnickles) [inline src/chi/components/sidebar/sidenav.scss]: Line 152 - Background color should be grey 100, not black

File: src/chi/components/sidebar/sidenav.scss [hunk 3]
```diff
@@ -376,6 +404,114 @@ $sidenav-lg-width: 12.5rem;
...
+            }
+          }
+        }
+      }
+
+      .m-sidenav__drawers {
+        .m-drawer {
+          &.-active {
+            .m-drawer__header {
+              background-color: set-color(grey, 100);
```
Comment (mattnickles) [inline src/chi/components/sidebar/sidenav.scss]: Background color should be grey 100 with a 50% alpha

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/541/ for more details.

File: src/chi/javascript/components/sidenav.js (hunk 1 above)
Comment (mattnickles) [reply src/chi/javascript/components/sidenav.js]: @janibolkvadze Collapsing effect is very slow. Can we eliminate that delay? We should also probably remove the `-active` class after collapsing.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/545/ for more details.

**Comment by mattnickles** [comment]
@janibolkvadze If Menu Item 1 is active and Menu Item 2 is clicked, I see we're applying the class `unselected` to Menu Item 1. However, if no secondary link within Menu Item 2 was clicked and the **drawer is closed**, we should probably remove the `unselected` class from Menu Item 1 so it appears as the active item again. What do you think?

**Comment by jllr** [comment]
@dani-cl-madrid can you take a look at this PR too? Thanks in advance

File: src/chi/components/sidebar/sidenav.scss [hunk 4]
```diff
@@ -116,6 +116,16 @@ $sidenav-lg-width: 12.5rem;
       &:not(.-active) > a:hover {
         background-color: set-color(grey, 10);
       }
+
+      .m-sidenav__drawer-item-list {
+        display: none;
+        overflow: hidden;
+        transition: height 0.075s ease-out;
+
+        &.visible {
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze we usually use - (dash) for this kind of classes (ie: -active). Please add it.

File: src/chi/components/sidebar/sidenav.scss [hunk 5]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+          ul {
+            &.m-sidenav__list {
+              li {
+                border-bottom: 0;
+
+                &.-active {
+                  a {
+                    background-color: rgba(set-color(mint, 40), 0.08);
+
+                    &.unselected {
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze we usually use - (dash) for this kind of classes (ie: -active). Please add it.

File: src/chi/components/sidebar/sidenav.scss [hunk 6]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+        .m-drawer {
+          background-color: set-color(grey, 90);
+
+          &.-active {
+            .m-drawer__header {
+              align-items: center;
+              background-color: rgba(set-color(grey, 100), 0.5);
+              color: set-color(grey, 40);
+              display: flex;
+              font-size: 16px;
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze please remove this property and add it to .a-drawer__title and convert it to rem

File: src/chi/components/sidebar/sidenav.scss [hunk 7]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+              height: 4rem;
+              padding-left: 1.5rem;
+
+              .a-drawer__title {
+                font-weight: 600;
+                left: 1.25rem;
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze this property is not necessary. Please remove it.

File: src/chi/components/sidebar/sidenav.scss [hunk 8]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+            .m-drawer__content {
+              background-color: set-color(grey, 90);
+
+              .m-sidenav__drawer-list {
+                border-top: 0;
+                padding-top: 0.5rem;
+
+                li {
+                  border-bottom: 0;
+                  line-height: 20px;
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze please convert it to rem.

File: src/chi/components/sidebar/sidenav.scss [hunk 9]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+
+                  .m-sidenav__drawer-item-list {
+                    background-color: set-color(grey, 80);
+
+                    .a-tabs {
+                      &.-sm {
+                        li {
+                          a {
+                            &.a-sidenav__drawer-item-tab {
+                              font-size: 13px;
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze please convert it to rem.

File: src/chi/components/sidebar/sidenav.scss [hunk 10]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+                              font-weight: 400;
+                              line-height: 20px;
```
Comment (jllr) [inline src/chi/components/sidebar/sidenav.scss]: @janibolkvadze please convert it to rem.

File: src/chi/components/sidebar/sidenav.scss [hunk 11]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+                }
+              }
+            }
+
+            .m-drawer__content {
+              background-color: set-color(grey, 90);
+
+              .m-sidenav__drawer-list {
+                border-top: 0;
+                padding-top: 0.5rem;
```
Comment (mattnickles) [inline src/chi/components/sidebar/sidenav.scss]: Let's remove this `padding-top: 0.5rem;`

File: src/chi/components/sidebar/sidenav.scss [hunk 12]
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
...
+                        li {
+                          a {
+                            &.a-sidenav__drawer-item-tab {
+                              font-size: 13px;
+                              font-weight: 400;
+                              line-height: 20px;
+                              padding-bottom: 0.375rem;
+                              padding-left: 3rem;
+                              padding-top: 0.375rem;
+                            }
```
Comment (mattnickles) [inline src/chi/components/sidebar/sidenav.scss]: In their design a hover state is defined on `m-sidenav__drawer-item-list` tab items. Let's add:
```
&:hover {
   background-color: set-color(grey, 70);
}
```

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/560/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/562/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/575/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/580/ for more details.

**Pull request merged**